mirte_robot/linetrace.py

```diff
@@ -16,11 +16,19 @@
 # Global shared memory objects (TODO: check if we need shared memory, why is server working?)
 stepper = multiprocessing.Value('b', True)
 do_step = multiprocessing.Value('b', False)
+running = multiprocessing.Value('b', False)
 
-def stop_mirte():
-     process.terminate()
+# the stop be stopped when:
+# 1) the code finishes
+# 2) the user stopped the process
+# 3) the websocket connection is closed
+def stop_mirte(terminate = True):
+     global running
+     if terminate:
+        process.terminate()
+     running.value = False
 
-def load_mirte_module(stepper, do_step):
```

Comment:
What is the running var for? It's not used in this function

Reply:
The running value is a replacement for process.is_alive(). Not sure anymore, but the process.is_alive() did not work correctly at that point.